Headers/DebugServer2/Utils/Enums.h

```diff
@@ -0,0 +1,48 @@
+//
+// Copyright (c) 2014-present, Facebook, Inc.
+// All rights reserved.
+//
+// This source code is licensed under the University of Illinois/NCSA Open
+// Source License found in the LICENSE file in the root directory of this
+// source tree. An additional grant of patent rights can be found in the
+// PATENTS file in the same directory.
+//
+
+template <typename Enum> struct EnableBitMaskOperators {
+  static const bool enable = false;
+};
+
+template <typename Enum>
+typename std::enable_if<EnableBitMaskOperators<Enum>::enable, Enum>::type
+operator|(Enum lhs, Enum rhs) {
+  using underlying = typename std::underlying_type<Enum>::type;
```

Comment:
`s/underlying/Underlying/`

Reply:
Additionally you're `using Underlying = typename std::underlying_type<Enum>::type;` in basically every one of these. Is there anyway you can pull it out and write it once? :P

Reply:
Indeed. You could clearly wrap this in more macros and what not but I'm not sure layering this in more levels of indirection makes it any clearer or better. I debated on a `unaryop` and `binaryop` template of some sort but meh.